Mathlib/CategoryTheory/Category/ULift.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright (c) 2021 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+
+! This file was ported from Lean 3 source module category_theory.category.ulift
+! leanprover-community/mathlib commit 32253a1a1071173b33dc7d6a218cf722c6feb514
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Category.Basic
+import Mathlib.CategoryTheory.Equivalence
+import Mathlib.CategoryTheory.EqToHom
+import Mathlib.Data.ULift
+
+/-!
+# Basic API for ULift
+
+This file contains a very basic API for working with the categorical
+instance on `ULift C` where `C` is a type with a category instance.
+
+1. `category_theory.ULift.up` is the functorial version of the usual `ULift.up`.
+2. `category_theory.ULift.down` is the functorial version of the usual `ULift.down`.
+3. `category_theory.ULift.equivalence` is the categorical equivalence between
+  `C` and `ULift C`.
+
+# ULiftHom
+
+Given a type `C : Type u`, `ULiftHom.{w} C` is just an alias for `C`.
+If we have `category.{v} C`, then `ULiftHom.{w} C` is endowed with a category instance
+whose morphisms are obtained by applying `ULift.{w}` to the morphisms from `C`.
+
+This is a category equivalent to `C`. The forward direction of the equivalence is `ULiftHom.up`,
+the backward direction is `ULiftHom.donw` and the equivalence is `ULiftHom.equiv`.
+
+# AsSmall
+
+This file also contains a construction which takes a type `C : Type u` with a
+category instance `category.{v} C` and makes a small category
+`AsSmall.{w} C : Type (max w v u)` equivalent to `C`.
+
+The forward direction of the equivalence, `C ⥤ AsSmall C`, is denoted `AsSmall.up`
+and the backward direction is `AsSmall.down`. The equivalence itself is `AsSmall.equiv`.
+-/
+
+universe w₁ v₁ v₂ u₁ u₂
+
+namespace CategoryTheory
+
+variable {C : Type u₁} [Category.{v₁} C]
+
+/-- The functorial version of `ULift.up`. -/
+@[simps]
+def Ulift.upFunctor : C ⥤ ULift.{u₂} C where
```

Comment:
I suppose all the `Ulift`'s in this file should be `ULift` to match the naming in `Data.ULift`.

Reply:
Done.